src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -132,17 +133,22 @@ class Simplex_tree {
 
   struct Key_simplex_base_real {
     Key_simplex_base_real() : key_(-1) {}
+    Key_simplex_base_real(Simplex_key k) : key_(k) {}
     void assign_key(Simplex_key k) { key_ = k; }
     Simplex_key key() const { return key_; }
    private:
     Simplex_key key_;
   };
   struct Key_simplex_base_dummy {
     Key_simplex_base_dummy() {}
+    Key_simplex_base_dummy(Simplex_key k) {
```

Comment:
I think the conclusion was to remove this constructor?

Reply:
Yes, it should be done now.